### PR TITLE
ICP with scale

### PR DESCRIPTION
**Pull request by DumDereDum**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by alalek** [comment]
> into `opencv:next`

wrong target branch

> do_not_target_to_shadow_branch

File: modules/3d/src/rgbd/odometry_functions.hpp [hunk 1]
```diff
@@ -100,7 +117,23 @@ typedef
...
+    C[1] = p0.z * n1[0] - p0.x * n1[2];
+    C[2] = -p0.y * n1[0] + p0.x * n1[1];
+    C[3] = n1[0];
+    C[4] = n1[1];
+    C[5] = n1[2];
+    double diff = n1.dot(p1 - p0);
+    if (diff < DBL_EPSILON || abs(diff) > 1000000)
+        C[6] = 0;
+    else
+        C[6] = n1.dot(p1-p0);
```
Comment (savuor) [inline modules/3d/src/rgbd/odometry_functions.hpp:132]: This should be checked if this formula is to be fixed or not:
it's possible that there should be `n1.dot((p0 - currentTranslation)/currentScale)`, it's easier to do at caller's side.

**Review by savuor — CHANGES_REQUESTED** [review]
Maybe I'll do these changes instead of @DumDereDum

**Comment by asmorkalov** [comment]
@DumDereDum Friendly reminder.

**Comment by savuor** [comment]
The algorithm does not converge even after a couple of fixes: the scale can drop to 0.02 or grow to 2.8 even when changing scale by 3%.
Since the code is not based on any known working solution, just on basic principles, it requires more deep research before making a PR.
But this would be easier to do in a different PR since it may require vast code changes.

**Pull request closed without merging**